src/core/infrastructure/BaseService.ts

```diff
@@ -34,6 +36,10 @@ export class BaseService {
     else if (jobToken) this.headers['job-token'] = jobToken;
     else if (token) this.headers['private-token'] = token;
 
+    // Profiling
+    if (profileToken) this.headers['X-Profile-Token'] = profileToken;
+    if (profileMode) this.headers['X-Profile-Mode'] = profileMode;
```

Comment:
nit: this will always be added, it should only be added if `x-profile-token` is set

Reply:
🤦‍♂ yup, overlooked that.

Reply:
Fixed